### PR TITLE
🧹 Sweep authSpecification in connectors repo

### PR DESCRIPTION
**Pull request by lmossman**
## What
Relates to https://github.com/airbytehq/airbyte-protocol/pull/39 and https://github.com/airbytehq/airbyte-platform-internal/pull/7595

This PR sweeps all occurrences of `authSpecification` from the `airbytehq/airbyte` repository as part of the [project](https://docs.google.com/document/d/1vBAHI9r5qcD8WJc0GXaWEsRrmrvKd4NNvmreUDrqzcA/edit?usp=sharing) to deprecate that field.

## How
Remove references to `authSpecification`, and update the tests as necessary to test the `advanced_auth` configuration instead.

Note: The legacy `authSpecification` still appears in the `cloud_registry.json`, `oss_registry.json`, and `connector_catalog.json` files, but this shouldn't matter since those are only used for testing general behavior that doesn't rely on this field, and will be wiped out the next time those files are refreshed due to a catalog shape change.

## Recommended reading order
1. `airbyte-api/src/main/openapi/config.yaml` - removes authSpecification from the api
2. `airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py` - updates the `test_oauth_flow_parameters` CAT test to verify that all paths in the `advanced_auth` object exist in the connector spec, rather than testing something similar for `authSpecification`
3. `airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_spec.py` - updates the CAT unit tests to account for the above change
4. The rest falls out of the above

## 🚨 User Impact 🚨
No breaking changes or user impact

### CODE REVIEW

**Comment by lmossman** [comment]
Note: keeping this in draft because there are still 2 connectors that need to be migrated off of `authSpecification`: `source-rd-station-marketing` and `destination-google-sheets`. This is planned to be done this week, after which this PR can be merged.

But it is ready for review now!

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/json_schema_helper.py
Comment (girarda) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/json_schema_helper.py:252]: nit: the pythonic way to do this is 
```
return [
        "/" + "/".join(value["path_in_connector_config"])
        for value in schema.values()
    ]
```

**Review by girarda — APPROVED** [review]
lgtm!

**Review by brianjlai — APPROVED** [review]
lgtm, but once the other two connectors are updated, just make sure to run the `legacy-test` command and verify the spec test changes still pass

**Comment by lmossman** [comment]
/legacy-test connector=integrations/source-rd-station-marketing

**Comment by lmossman** [comment]
/legacy-test connector=integrations/destination-google-sheets

**Comment by lmossman** [comment]
/legacy-test connector=connectors/source-rd-station-marketing

**Comment by octavia-squidington-iii** [comment]
##  destination-mssql-strict-encrypt test report (commit [`235f335240`](https://github.com/airbytehq/airbyte/pull/27996/commits/235f33524066cb83c906904c2b2cb80a0a14bbc6)) - ❌

⏲️  Total pipeline duration: 31mn55s 

| Step                                                                                        | Result   |
|:--------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mssql-strict-encrypt/metadata.yaml     | ✅        |
| Connector version semver check                                                              | ✅        |
| QA checks                                                                                   | ✅        |
| Build connector tar                                                                         | ✅        |
| Build destination-mssql-strict-encrypt docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-mssql:dev                                                       | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mssql-strict-encrypt:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690331632/235f33524066cb83c906904c2b2cb80a0a14bbc6/destination-mssql-strict-encrypt/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mssql-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`235f335240`](https://github.com/airbytehq/airbyte/pull/27996/commits/235f33524066cb83c906904c2b2cb80a0a14bbc6)) - ❌

⏲️  Total pipeline duration: 61mn41s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ❌        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690331632/235f33524066cb83c906904c2b2cb80a0a14bbc6/source-facebook-marketing/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

File: airbyte-integrations/connectors/source-google-sheets/metadata.yaml
Comment (lmossman) [inline airbyte-integrations/connectors/source-google-sheets/metadata.yaml:16]: Unpinning because this no longer needs to be pinned:
> [This PR added a field to the spec](https://github.com/airbytehq/airbyte/pull/23057/files#diff-7713475557366d1d512b0822754fe63fc44ea9fc3fd85d274f64965a741a1487), but it defaults to [false](https://github.com/airbytehq/airbyte/pull/23057/files#diff-7713475557366d1d512b0822754fe63fc44ea9fc3fd85d274f64965a741a1487R27), so the [default behavior will stay the same](https://github.com/airbytehq/airbyte/pull/23057/files#diff-c397937888441dc0f05d5a3a8590cabbaba287b0f305001c89ab72df9fc4100fR147-R148) :+1:
[The major bump is just a change in license, no change in behavior](https://github.com/airbytehq/airbyte/pull/27738/files) :+1:
I don't see [outstanding issues for source-google-sheets](https://github.com/airbytehq/airbyte/issues?q=is%3Aopen+is%3Aissue+label%3Aconnectors%2Fsource%2Fgoogle-sheets), so I expect the latest version of the connector to work as expected

**Comment by octavia-squidington-iii** [comment]
##  source-google-sheets test report (commit [`4a754eace5`](https://github.com/airbytehq/airbyte/pull/27996/commits/4a754eace5b69d75e0ccb4b2f2b43104372589c6)) - ❌

⏲️  Total pipeline duration: 01mn45s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-sheets/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| QA checks                                                                   | ✅        |
| Code format checks                                                          | ✅        |
| Connector package install                                                   | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690402484/4a754eace5b69d75e0ccb4b2f2b43104372589c6/source-google-sheets/0.3.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-sheets test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`4a754eace5`](https://github.com/airbytehq/airbyte/pull/27996/commits/4a754eace5b69d75e0ccb4b2f2b43104372589c6)) - ✅

⏲️  Total pipeline duration: 66mn12s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690402484/4a754eace5b69d75e0ccb4b2f2b43104372589c6/source-facebook-marketing/1.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-mariadb-columnstore test report (commit [`4a754eace5`](https://github.com/airbytehq/airbyte/pull/27996/commits/4a754eace5b69d75e0ccb4b2f2b43104372589c6)) - ❌

⏲️  Total pipeline duration: 100mn54s 

| Step                                                                                       | Result   |
|:-------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mariadb-columnstore/metadata.yaml     | ✅        |
| Connector version semver check                                                             | ✅        |
| QA checks                                                                                  | ✅        |
| Build connector tar                                                                        | ✅        |
| Build destination-mariadb-columnstore docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mariadb-columnstore:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690402484/4a754eace5b69d75e0ccb4b2f2b43104372589c6/destination-mariadb-columnstore/0.1.7/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mariadb-columnstore test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-exasol test report (commit [`4a754eace5`](https://github.com/airbytehq/airbyte/pull/27996/commits/4a754eace5b69d75e0ccb4b2f2b43104372589c6)) - ❌

⏲️  Total pipeline duration: 15mn00s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-exasol/metadata.yaml     | ✅        |
| Connector version semver check                                                | ✅        |
| QA checks                                                                     | ❌        |
| Build connector tar                                                           | ✅        |
| Build destination-exasol docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-exasol:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690402484/4a754eace5b69d75e0ccb4b2f2b43104372589c6/destination-exasol/0.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-exasol test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-pubsub test report (commit [`4a754eace5`](https://github.com/airbytehq/airbyte/pull/27996/commits/4a754eace5b69d75e0ccb4b2f2b43104372589c6)) - ✅

⏲️  Total pipeline duration: 18mn35s 

| Step                                                                          | Result   |
|:------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-pubsub/metadata.yaml     | ✅        |
| Connector version semver check                                                | ✅        |
| QA checks                                                                     | ✅        |
| Build connector tar                                                           | ✅        |
| Build destination-pubsub docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-pubsub:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690402484/4a754eace5b69d75e0ccb4b2f2b43104372589c6/destination-pubsub/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pubsub test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-databricks test report (commit [`4a754eace5`](https://github.com/airbytehq/airbyte/pull/27996/commits/4a754eace5b69d75e0ccb4b2f2b43104372589c6)) - ❌

⏲️  Total pipeline duration: 55mn16s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-databricks/metadata.yaml     | ✅        |
| Connector version semver check                                                    | ✅        |
| QA checks                                                                         | ✅        |
| Build connector tar                                                               | ✅        |
| Build destination-databricks docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-databricks:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690402484/4a754eace5b69d75e0ccb4b2f2b43104372589c6/destination-databricks/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-databricks test
```

**Comment by octavia-squidington-iii** [comment]
##  source-google-sheets test report (commit [`bbb38c11a7`](https://github.com/airbytehq/airbyte/pull/27996/commits/bbb38c11a7277be93651b36ac8eb8cbc45773bd3)) - ❌

⏲️  Total pipeline duration: 05mn02s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-sheets/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ❌        |
| QA checks                                                                   | ✅        |
| Code format checks                                                          | ✅        |
| Connector package install                                                   | ✅        |
| Build source-google-sheets docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                  | ✅        |
| Acceptance tests                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690489176/bbb38c11a7277be93651b36ac8eb8cbc45773bd3/source-google-sheets/0.3.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-sheets test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-r2 test report (commit [`bbb38c11a7`](https://github.com/airbytehq/airbyte/pull/27996/commits/bbb38c11a7277be93651b36ac8eb8cbc45773bd3)) - ❌

⏲️  Total pipeline duration: 08mn06s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-r2/metadata.yaml     | ✅        |
| Connector version semver check                                            | ✅        |
| QA checks                                                                 | ✅        |
| Build connector tar                                                       | ✅        |
| Build destination-r2 docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-r2:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690489176/bbb38c11a7277be93651b36ac8eb8cbc45773bd3/destination-r2/0.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-r2 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-snowflake test report (commit [`bbb38c11a7`](https://github.com/airbytehq/airbyte/pull/27996/commits/bbb38c11a7277be93651b36ac8eb8cbc45773bd3)) - ❌

⏲️  Total pipeline duration: 10mn27s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                              | ✅        |
| QA checks                                                                   | ✅        |
| Build connector tar                                                         | ✅        |
| Build source-snowflake docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-snowflake:integrationTest | ❌        |
| Acceptance tests                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690489176/bbb38c11a7277be93651b36ac8eb8cbc45773bd3/source-snowflake/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  source-postgres test report (commit [`bbb38c11a7`](https://github.com/airbytehq/airbyte/pull/27996/commits/bbb38c11a7277be93651b36ac8eb8cbc45773bd3)) - ✅

⏲️  Total pipeline duration: 14mn30s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-postgres/metadata.yaml     | ✅        |
| Connector version semver check                                             | ✅        |
| QA checks                                                                  | ✅        |
| Build connector tar                                                        | ✅        |
| Build source-postgres docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-postgres:integrationTest | ✅        |
| Acceptance tests                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690489176/bbb38c11a7277be93651b36ac8eb8cbc45773bd3/source-postgres/3.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`bbb38c11a7`](https://github.com/airbytehq/airbyte/pull/27996/commits/bbb38c11a7277be93651b36ac8eb8cbc45773bd3)) - ❌

⏲️  Total pipeline duration: 61mn49s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690489176/bbb38c11a7277be93651b36ac8eb8cbc45773bd3/source-facebook-marketing/1.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-approvington** [comment]
Our work here is done <br/> done

**Review by octavia-approvington — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  destination-mssql test report (commit [`bbb38c11a7`](https://github.com/airbytehq/airbyte/pull/27996/commits/bbb38c11a7277be93651b36ac8eb8cbc45773bd3)) - ❌

⏲️  Total pipeline duration: 304mn01s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-mssql/metadata.yaml     | ✅        |
| Connector version semver check                                               | ✅        |
| QA checks                                                                    | ✅        |
| Build connector tar                                                          | ✅        |
| Build destination-mssql docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-mssql:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-mssql:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/lmossman_deprecate-auth-specification/1690489176/bbb38c11a7277be93651b36ac8eb8cbc45773bd3/destination-mssql/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-mssql test
```

**Comment by bazarnov** [comment]
@lmossman Hello! It seems like this change break some more `source` connectors, like:

- source-github - [test](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/nightly_builds/master/1690762548/ad12b1e1b20713482ec2e6c3b6e8ae95b86e6dbd/source-github/1.0.2/output.html)
- source-instagram - [test](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/nightly_builds/master/1690762548/ad12b1e1b20713482ec2e6c3b6e8ae95b86e6dbd/source-instagram/1.0.9/output.html)
- source-zendesk-support - [test](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/nightly_builds/master/1690762548/ad12b1e1b20713482ec2e6c3b6e8ae95b86e6dbd/source-zendesk-support/0.10.4/output.html)
- source-zendesk-talk - [test](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/nightly_builds/master/1690762548/ad12b1e1b20713482ec2e6c3b6e8ae95b86e6dbd/source-zendesk-talk/0.1.7/output.html)

They are failed on CAT with this:

```
AssertionError: Specified oauth fields are missed from spec schema: {'/credentials/client_secret', '/credentials/client_id'}
E       assert equals failed
E         set(^[                       set(^)                      
E         - '/credentials/client_id',                             
E         - '/credentials/client_secr                             
E         -et',                                                   
E         -])
```

Assuming that `client_id` and `client_secret` fields are available in the `oneOf / or root` connector's `spec`. However this was not an issue 3 days ago, I wonder why this is happening now, could you please advice?

**Comment by bazarnov** [comment]
UPDATED:

Apparently some sources like `Trello` used to return some `server_input_configuration` values to the:
```json
"complete_oauth_server_output_specification": {
        "type": "object",
        "additionalProperties": false,
        "properties": {
          "client_id": {
            "type": "string",
            "path_in_connector_config": ["client_id"]
          },
          "client_secret": {
            "type": "string",
            "path_in_connector_config": ["client_secret"]
          }
        }
      }
```
Removing this `complete_oauth_server_output_specification` resolved the CAT issues. because there is nowhere to return the values for the input config, and they are not used elsewhere but for OAuth Flow server input. 

I guess the rest of the failed sources CATs died for the same reason, fixing them now.

**Comment by lmossman** [comment]
@bazarnov that sounds right to me; for example, looking at `source-github`'s spec, it looks like this field is pointing at the path [`credentials`, `client_id`], but that path doesn't actually exist in the connectionSpecification

**Comment by bazarnov** [comment]
it is, however the Connectors Team was not aware of this change) It would be very helpful if we know that prior to the merge, if it's possible)

**Comment by lmossman** [comment]
Yes I definitely could have communicated this better, sorry about that!

I didn't expect this change to cause new failures in connectors since it should have been testing the same type of things as the old authSpecification test, but in these cases the connectors failed the new test because of an underlying bug in their spec

**Pull request merged**
